ts/parts/Series.ts

```diff
@@ -5189,11 +5189,13 @@ H.Series = seriesType<Highcharts.LineSeries>(
                     pointStack,
                     stackValues;
 
-                // Discard disallowed y values for log axes (#3434)
-                if (yAxis.positiveValuesOnly &&
-                    yValue !== null &&
-                    (yValue as any) <= 0
-                ) {
+                // Discard disallowed x and y values for log axes #3434, #13533
+                const isPositiveAxes = yAxis.positiveValuesOnly ||
+                    xAxis.positiveValuesOnly,
+                    isValidValue = (yValue !== null && (yValue as any) <= 0) ||
+                    (xValue && xValue <= 0);
+
+                if (isPositiveAxes && isValidValue) {
```

Comment:
We need to make sure that only nulls/not positive values should set `isNull` flag. Correct my if I'm wrong but this case will be wrong:

- xAxis: log, yAxis: linear -> `isPositiveAxes = true`
- xValue: 1, yValue: 0 -> `isValidValue=true` 

Which means we will set `isNull` to true, but we shouldn't. It's a valid point.

`isValidValue` - that variable name is a bit misleading.

Perhaps it's a good idea, to create `Axis.proto.validateValue(value)` method and call it here and in above `getSeriesExtremes()` (instead of two filters: `isPositive` and isNumber`).